misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/mock/ObjectMock.java

```diff
@@ -315,28 +641,49 @@ private static boolean validateFieldAccessIsPublic(ObjectType objectType, String
         BObject mockObj = genericMock.getMockObj();
-        String functionName = caseObj.getStringValue(StringUtils.fromString("functionName")).toString();
-        BArray returnVals = caseObj.getArrayValue(StringUtils.fromString("returnValueSeq"));
-
-        for (int i = 0; i < returnVals.getValues().length; i++) {
-            if (returnVals.getValues()[i] == null) {
+        String functionName = caseObj.getStringValue(StringUtils.fromString(MockConstants.FUNCTION_NAME)).toString();
+        BArray returnValueSequence = caseObj.getArrayValue(StringUtils.fromString(MockConstants.RETURN_VALUE_SEQ));
+        for (int i = 0; i < returnValueSequence.getValues().length; i++) {
+            Object returnValue = returnValueSequence.getValues()[i];
+            if (returnValue == null) {
                 break;
             }
-            if (!validateReturnValue(functionName, returnVals.getValues()[i],
-                    ((ObjectType) TypeUtils.getImpliedType(genericMock.getType())).getMethods())) {
-                String detail = "return value provided at position '" + i
-                        + "' does not match the return type of function '" + functionName + "()'";
-                return ErrorCreator.createError(
-                        MockConstants.TEST_PACKAGE_ID,
-                        MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
-                        StringUtils.fromString(detail),
-                        null,
-                        new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
+            if (MockConstants.MEMBER_RESOURCE_FUNCTION_STUB.equals(mockType)) {
+                String functionPattern = getFunctionNameForResourcePath(functionName);
+                String accessor = caseObj.getStringValue(StringUtils.fromString(MockConstants.ACCESSOR)).toString();
+                String resourceFunctionPattern = MockConstants.DOLLAR_RESOURCE_SEPARATOR + accessor +
+                        MockConstants.DOLLAR_RESOURCE_SEPARATOR + functionPattern;
+                if (!validateReturnValueForResourcePath(resourceFunctionPattern, returnValue,
+                        ((BClientType) objectType).getResourceMethods())) {
+                    String detail = "return value provided at position '" + i
+                            + "' does not match the return type of the resource path '" + functionName + "'";
+                    return ErrorCreator.createError(
+                            MockConstants.TEST_PACKAGE_ID,
+                            MockConstants.FUNCTION_SIGNATURE_MISMATCH_ERROR,
+                            StringUtils.fromString(detail),
+                            null,
+                            new MapValueImpl<>(PredefinedTypes.TYPE_ERROR_DETAIL));
+                }
+                MockRegistry.getInstance().registerCase(mockObj, resourceFunctionPattern, null, returnValue, i + 1);
+            } else {
+                if (!validateReturnValue(functionName, returnValue,
```

Comment:
let's return early by inverting?

Reply:
I think here we do the same, we check the return value. If the return value does not match then, we return.